src/Compilers/CSharp/Portable/FlowAnalysis/DataFlowPass.cs

```diff
@@ -1630,16 +1630,6 @@ public override BoundNode VisitUsingStatement(BoundUsingStatement node)
 
```

Comment:
This has grown in complexity over time. It used to be a simple assertion that the locals were all using variables. But now out variables and pattern variables can occur here too, and out variables were designed in a way that makes them indistinguishable from other locals. Since they are declared here, we need to allocate slots for them. We don't actually need the switch except to support the assertion.

The "conditional" call to base.Visit... is just a shorthand way of skipping the following foreach loop... and the one after that. It could also be done as an if statement around them, but that seemed slightly more clumsy.

---
In reply to: [78859460](https://github.com/dotnet/roslyn/pull/13807#discussion_r78859460) [](ancestors = 78859460)